pkg/webhook/subscription.go

```diff
@@ -0,0 +1,64 @@
+package webhook
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/go-logr/logr"
+	opv1a1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	"github.com/red-hat-storage/ocs-client-operator/api/v1alpha1"
+	"github.com/red-hat-storage/ocs-client-operator/pkg/utils"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+type SubscriptionAdmission struct {
+	Client  client.Client
+	Decoder *admission.Decoder
+	Log     logr.Logger
+}
+
+func (s *SubscriptionAdmission) Handle(ctx context.Context, req admission.Request) admission.Response {
+	s.Log.Info("Request received for admission review")
+
+	// review should be for a subscription
+	subscription := &opv1a1.Subscription{}
+	if err := s.Decoder.Decode(req, subscription); err != nil {
+		s.Log.Error(err, "failed to decode admission review as subscription")
+		return admission.Errored(http.StatusBadRequest, fmt.Errorf("only subscriptions admission reviews are supported: %v", err))
+	}
+
+	// review should be for ocs-client-operator subscription
+	if subscription.Spec.Package != "ocs-client-operator" {
+		s.Log.Info("subscription package is not 'ocs-client-operator'")
+		return admission.Errored(http.StatusBadRequest, fmt.Errorf("only ocs-client-operator subscription validation is supported"))
+	}
+
+	storageClients := &v1alpha1.StorageClientList{}
+	if err := s.Client.List(ctx, storageClients); err != nil {
+		return admission.Errored(http.StatusInternalServerError, fmt.Errorf("failed to list storageclients for validating subscription request: %v", err))
+	}
+
+	// this is the channel that user want to subscribe to
+	requestedUserSubcriptionChannel := subscription.Spec.Channel
```

Comment:
In my prev comment I suggested: `requestedSubcriptionChannel` not `requestedUserSubcriptionChannel`
The reason was that we don't know what initiated the change and we cannot assume it was a "User" 
```suggestion
	requestedSubcriptionChannel := subscription.Spec.Channel
```